### ISSUE TITLE
Load folders only on unfolding tree nodes

### ISSUE THREAD

**Issue by TomBraider42** [pohmelie/cuda_project_man#38]
To open a project with near 5000 files takes a lot of time.
Could you speed up the tree_proc( TREE_ITEM_ADD ) function or use a cache file?
So we can use the refresh function to reload the project only if needed.

**Comment by Alexey-T** [comment]
tree_proc(...._ADD) uses native Lazarus function, to add node to Treeview, I dont know how change it
```
  TREE_ITEM_ADD:
          begin
            Node1:= Tree.Items.AddChild(TreeNode, StrText);
            Node1.Text:= StrText;
            Node1.ImageIndex:= NImageIndex;
            Node1.SelectedIndex:= NImageIndex;
            if NIndex>=0 then
              Node1.Index:= NIndex;       
```

**Comment by TomBraider42** [comment]
Maybe it's possible to parse/open only the root directory on start.
Adding files of a subdirectory just after open it in tree panel.
Same as in FTP Tree Panel.

**Comment by pohmelie** [comment]
@TomBraider42, this sounds right.

**Comment by Alexey-T** [comment]
I should add on_unfold event for Treeview. with it itis possible then. todo.

**Comment by Alexey-T** [comment]
It is possible to handle on_sel for treeview. in on_sel possible to load data for node. On showing root node must load, next nodes must load in on_sel

@pohmelie Are you busy?

**Comment by pohmelie** [comment]
@Alexey-T 
Here is recursion part of treeview refresh
https://github.com/pohmelie/cuda_project_man/blob/master/__init__.py#L434
I will try make changes to add this behavior.

**Comment by Alexey-T** [comment]
Good. I dont understand pathlib yet... and iterdir..

**Comment by pohmelie** [comment]
@Alexey-T, it looks like it works now as expected. Check it out.
added:
Actually directory reveals on double click. I think it is more expectable for user.

**Comment by pohmelie** [comment]
But, probably, we need toggle for this. I mean eager and lazy revealing of directories.

**Comment by Alexey-T** [comment]
On double-click and on click on +- sign, and on usual press Num+, needed on any unfold.

**Comment by pohmelie** [comment]
No events triggered on num+ or +/-.

**Comment by Alexey-T** [comment]
No events, yes, may you use on_sel?

**Comment by pohmelie** [comment]
on_sel for what? Unfold for on_sel looks like it not what user expect.

**Comment by Alexey-T** [comment]
Let's wait until on-unfold will be

**Comment by Alexey-T** [comment]
@TomBraider42 It's solved in the Cud 1.10.0. Pls try update [included  into Cud]
